demux/demux_playlist.c

```diff
@@ -175,7 +178,25 @@ static char *pl_get_line0(struct pl_parser *p)
 
 static bstr pl_get_line(struct pl_parser *p)
 {
-    return bstr0(pl_get_line0(p));
+    bstr line = bstr_strip(bstr0(pl_get_line0(p)));
+    const char *charset = mp_charset_guess(p, p->log, line, p->codepage, 0);
```

Comment:
Applying charset guessing for every line read seems not ideal

Reply:
`mp_charset_guess` will internally check if it's valid UTF-8 first before actually guessing a charset (unless you force a codepage of course).

Reply:
sure but if the guessing *is* happening it will happen on every line in isolation, no?

Reply:
Sure but how else would you do it? I guess you could read the whole file first and convert it if needed but that doesn't really seem better since you have to go back and parse line-by-line anyway.

Reply:
I thought more about this and you're right, the alternatives are not better.